src/interactive-auth.ts

```diff
@@ -156,6 +156,11 @@ interface IOpts {
      */
     emailSid?: string;
 
+    /**
+     * If specified will prefer flows which entirely consist of supported stages to avoid needing the fallback mechanism
+     */
+    supportedStages?: string[];
```

Comment:
```suggestion
    supportedStages?: Array<AuthType | string>;
```

Because the app may support custom types which the js-sdk does not know about

Reply:
Fair enough. It might be nice to say in the comment "These should normally be `AuthType`s, but may be strings if the application supports custom stages"?